include/platform/mir/console_services.h

```diff
@@ -152,6 +152,10 @@ class ConsoleServices
         int major, int minor,
         std::unique_ptr<Device::Observer> observer) = 0;
 
+    virtual void register_lock_handler(
+        std::function<void()> const& lock,
+        std::function<void()> const& unlock) = 0;
```

Comment:
This is two-stage initialization (and only needed by one of multiple subclasses).

A cleaner design is to have a `ConsoleLocker` interface (implemented by `SessionLocker`) and pass that to the `LogindConsoleServices` constructor.

Reply:
You've acked this but not changed it.

Reply:
Oh I see now